dhall/src/Dhall/TypeCheck.hs

```diff
@@ -493,7 +493,7 @@ typeWithA tpa = loop
                 s0 <- fmap Dhall.Core.normalize (loop ctx t0)
                 c <- case s0 of
                     Const c -> pure c
-                    _       -> Left (TypeError ctx e (InvalidField k0 v0))
+                    _       -> Left (TypeError ctx e (InvalidFieldType k0 v0))
```

Comment:
It's probably off-topic, but can you give an example that would reach this branch?

Reply:
None that I know of, but I always handle all possible cases because an impossible branch today could become a possible branch tomorrow